src/loader/networks/sdex-chain.ts

```diff
@@ -28,7 +45,182 @@ export class SdexChain {
         quote: string;
         poolIdx: number;
       }[];
-    }>(gglPools, { limit });
+    }>(gqlPools, { limit });
+  }
+
+  public async queryUserPositions(user: string) {
+    return this.queryFromSubgraph<LiquidityChangesResponse>(gqlLiquidityChanges, { user });
+  }
+
+  private async executeMulticall(calls: Array<{ target: string; callData: string }>) {
+    const results = await this.multicall.tryAggregate.staticCall(true, calls);
+    if (results[0].success && results[0].returnData) {
+      return results[0].returnData;
+    } else {
+      throw new Error('Multicall failed or returned empty');
+    }
+  }
+
+  public async queryConcRewards(
+    owner: string,
+    base: string,
+    quote: string,
+    poolIdx: number,
+    lowerTick: number,
+    upperTick: number,
+  ) {
+    this.validateInputs(owner, base, quote, poolIdx);
+
+    const callData = this.queryContract.interface.encodeFunctionData('queryConcRewards', [
+      owner,
+      base,
+      quote,
+      poolIdx,
+      lowerTick,
+      upperTick,
+    ]);
+
+    const returnData = await this.executeMulticall([{ target: this.config.query, callData }]);
+    const result = this.queryContract.interface.decodeFunctionResult('queryConcRewards', returnData);
+    return parseRewardResult(result);
```

Comment:
You are making single call here, multicall is not needed (and even makes it work slower).

```suggestion
    return this.queryContract.staticCall.queryConcRewards(
      owner,
      base,
      quote,
      poolIdx,
      lowerTick,
      upperTick,
    );
```

I think idea would be just build list of datas with encodeFunctionData here, then put entire array to multicall in another function.